ward/testing.py

```diff
@@ -128,6 +128,10 @@ def run(self, cache: FixtureCache, idx: int = 0) -> "TestResult":
                     cache, iteration=self.param_meta.instance_index
                 )
                 self.format_description(resolved_args)
+                if dry_run:
```

Comment:
As it stands, `dry_run=True` will result in fixtures still being resolved, which I'm not sure is the desired behaviour. IMO, the `if dry_run` block should live right next to the `if isinstance(self.marker, SkipMarker)` block on line 118. Any thoughts?

To me, a dry-run wouldn't execute any fixtures. Of course, fixture execution is required to build a full description (replacing any variables used in format string descriptions).

Reply:
I originally had it after the SkipMarker check but found the formatted descriptions to be too useful. I think fixtures are not part of the test itself and therefore it is ok to resolve them during a dry-run. Maybe it is a good idea to inform users about broken fixtures, when they dry-run their tests.

Reply:
My main concern is that the use-case I had in mind for this feature was a way of accessing Ward's test discovery logic for use in editor plugins etc. Fixtures can execute arbitrary code, and so could include network calls and side-effects. Loading your text editor would result in potential network calls, rows being added in databases, and anything else your fixtures. I can't see anyone expecting opening their text editor to start executing their fixtures. Did you have a different use-case in mind?

I imagine the functionality provided by this PR would be useful in some cases, but not for the use-case I had in mind.

Reply:
Yeah that is a valid point.I Did not think about that.